tests/remote_lab/connect_vpn.pm

```diff
@@ -0,0 +1,38 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Connect VPN to a remote lab using openconnect
+# Maintainer: Oliver Kurz <okurz@suse.de>
+# Tags: https://progress.opensuse.org/issues/49901
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    my ($self) = @_;
+    my $vpn_username = get_required_var('VPN_USERNAME');
+    my $vpn_endpoint = get_var('VPN_ENDPOINT', 'asa003b.centers.ihost.com');
+    my $vpn_group = get_var('VPN_GROUP', 'ACC');
+    # nohup should already go to background but during test development I
+    # observed that it still blocked the terminal – regardless of e.g. using a
+    # virtio serial terminal or VNC based – so let's force it to the
+    # background.
+    assert_script_run "(echo \$vpn_password | nohup openconnect --user=$vpn_username --passwd-on-stdin --authgroup=$vpn_group $vpn_endpoint > vpn.log &)";
```

Comment:
`$vpn_password` is not defined

Reply:
well, it's `\$vpn_password` so defined in the shell context by the corresponding `read -s …` command in setup_vpn.pm

Reply:
Please add a comment then of what's going on :)

Reply:
done

Reply:
> done

Not done. Please be more explicit in the comment, if possible mention also the test module, because again: Being explicit is better than asuming that everybody is on the same page.

Reply:
Sorry but it seems like you missed the comment. It's in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7959/files#diff-4a676367e60346cf72f6392d62989750R25 only that the following four lines describe the use of the subshell around nohup. I can turn that around of course if you prefer?

Reply:
I'd prefer the shifting around, but will leave that to you. But yep, looks like I missed the last part of the comment ;)

Reply:
no problem, done